cmd/object-api-utils.go

```diff
@@ -64,12 +69,13 @@ func IsValidBucketName(bucket string) bool {
 	if len(bucket) < 3 || len(bucket) > 63 {
 		return false
 	}
-	if bucket[0] == '.' || bucket[len(bucket)-1] == '.' {
```

Comment:
The optimization is needed for the purpose i explained below, memory related issues with regexp.  One more thing we should do is avoid using 

```
validBucket.Match([]byte(string))
```